src/query/interpret/eval.hpp

```diff
@@ -1105,8 +1105,15 @@ class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
     }
   }
 
-  TypedValue Visit(PatternComprehension & /*pattern_comprehension*/) override {
-    throw utils::NotYetImplemented("Expression evaluator can not handle pattern comprehension.");
+  TypedValue Visit(PatternComprehension &pattern_comprehension) override {
+    TypedValue &frame_pattern_comprehension_value = frame_->at(symbol_table_->at(pattern_comprehension));
```

Comment:
I think you can put const here, and it will then be more obvious later we are returning copy of that value
```suggestion
    TypedValue const &frame_pattern_comprehension_value = frame_->at(symbol_table_->at(pattern_comprehension));
```